mq/publishers.go

```diff
@@ -176,3 +181,67 @@ func ServerStartNotify() error {
 	}
 	return nil
 }
+
+// function to collect and store metrics for server nodes
+func collectServerMetrics(networks []models.Network) {
+	if !logic.Is_EE {
+		return
+	}
+	if len(networks) > 0 {
+		for i := range networks {
+			currentNetworkNodes, err := logic.GetNetworkNodes(networks[i].NetID)
+			if err != nil {
+				continue
+			}
+			currentServerNodes := logic.GetServerNodes(networks[i].NetID)
+			if len(currentServerNodes) > 0 {
+				for i := range currentServerNodes {
+					if logic.IsLocalServer(&currentServerNodes[i]) {
+						serverMetrics := logic.CollectServerMetrics(currentServerNodes[i].ID, currentNetworkNodes)
+						if serverMetrics != nil {
+							serverMetrics.NodeName = currentServerNodes[i].Name
+							serverMetrics.NodeID = currentServerNodes[i].ID
+							serverMetrics.IsServer = "yes"
+							serverMetrics.Network = currentServerNodes[i].Network
+							if err = metrics.GetExchangedBytesForNode(&currentServerNodes[i], serverMetrics); err != nil {
+								logger.Log(1, fmt.Sprintf("failed to update exchanged bytes info for server: %s, err: %v",
+									currentServerNodes[i].Name, err))
+							}
+							updateNodeMetrics(&currentServerNodes[i], serverMetrics)
+							if err = logic.UpdateMetrics(currentServerNodes[i].ID, serverMetrics); err != nil {
+								logger.Log(1, "failed to update metrics for server node", currentServerNodes[i].ID)
+							}
+							if servercfg.IsMetricsExporter() {
+								logger.Log(2, "-------------> SERVER METRICS: ", fmt.Sprintf("%+v", serverMetrics))
+								if err := pushMetricsToExporter(*serverMetrics); err != nil {
+									logger.Log(2, "failed to push server metrics to exporter: ", err.Error())
+								}
+							}
+
+						}
+
+					}
+				}
+			}
+		}
+	}
+}
+
+func pushMetricsToExporter(metrics models.Metrics) error {
+	logger.Log(2, "----> Pushing metrics to exporter")
+	SetupMQTT()
```

Comment:
I guess we can remove `SetupMQTT()` here since mqclient is a global variable now